src/server/server.ts

```diff
@@ -314,6 +462,38 @@ function compileCommon(req: express.Request, res: express.Response, wrapWithRunn
         });
 }
 
+function parseXmlString(xml): Promise<JSON> {
+    return new Promise((resolve, reject) => {
+        xml2js.parseString(xml, (err, result) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(result);
+            }
+        });
+    });
+};
+
+function getVersionNumber(): Promise<string> {
+     return new Promise((resolve, reject) => {
+        return Request.post({
+            url: SCRIPT_LAB_STORE_URL,
+            body: SCRIPT_LAB_STORE_ID,
+        }, (error, httpResponse, body) => {
+            if (error) {
+                ai.trackException('[Snippet] Get version number failed', error);
```

Comment:
You could cache the previous version number retrieved from the web service for this error case and just reuse it if we fail to get the latest version number.

Reply:
Added.